src/transport/validator/uattributesvalidator.rs

```diff
@@ -25,169 +26,177 @@ use std::time::SystemTime;
 /// `UAttributesValidator` is a trait implemented by all validators for `UAttributes`. It provides functionality
 /// to help validate that a given `UAttributes` instance is correctly configured to define the Payload.
 pub trait UAttributesValidator {
-    fn validate(&self, attributes: &UAttributes) -> UStatus {
+    /// Takes a `UAttributes` object and runs validations.
+    ///
+    /// # Arguments
+    /// * `attributes` - The `UAttributes` to validate.
+    ///
+    /// # Returns
+    /// Returns a `UStatus` that indicates success or failure. If failed, it includes a message containing
+    /// all validation errors for invalid configurations.
+    fn validate(&self, attributes: &UAttributes) -> ValidationResult {
         let error_messages: Vec<String> = vec![
             self.validate_type(attributes),
-            self.validate_id(attributes),
+            self.validate_ttl(attributes),
             self.validate_sink(attributes),
             self.validate_commstatus(attributes),
-            self.validate_ttl(attributes),
+            self.validate_permission_level(attributes),
             self.validate_reqid(attributes),
         ]
         .into_iter()
-        .filter(|status| status.is_failed())
-        .map(|status| status.message().clone())
+        .filter(|status| status.is_failure())
+        .map(|status| status.get_message())
         .collect();
 
         let error_message = error_messages.join(", ");
         if error_message.is_empty() {
-            UStatus::ok()
+            ValidationResult::Success
         } else {
-            UStatus::fail_with_msg_and_reason(&error_message, UCode::InvalidArgument)
+            ValidationResult::Failure(error_message)
         }
     }
 
     fn type_name(&self) -> &'static str;
 
-    /// Validates the `MessageType` of `UAttributes`.
+    /// Indicates whether the payload with these [`UAttributes`] has expired.
     ///
-    /// # Arguments
+    /// # Parameters
     ///
-    /// * `attributes` - `UAttributes` object containing the message type to validate.
+    /// * `attributes`: Reference to a [`UAttributes`] struct containing the time-to-live value.
     ///
     /// # Returns
     ///
-    /// Returns a `UStatus` indicating whether the `MessageType` is valid or not.
-    fn validate_type(&self, attributes: &UAttributes) -> UStatus;
+    /// Returns a `ValidationResult` that is success or failed with a failure message.
+    fn is_expired(&self, attributes: &UAttributes) -> ValidationResult {
+        let ttl = match attributes.ttl {
+            Some(t) if t > 0 => t,
+            Some(_) => return ValidationResult::Success,
+            None => 0,
+        };
+
+        if let Some(uuid) = &attributes.id {
+            if let Some(time) = UuidUtils::get_time(uuid) {
+                let delta = SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
+                    .unwrap()
```

Comment:
The `unwrap()` could be replace with an early return into `ValidationResult`. In that way a panic can be avoided (unless a panic is the desired behaviour`).

Reply:
done